provisionersdk/scripts/bootstrap_linux.sh

```diff
@@ -86,4 +86,12 @@ fi
 
 export CODER_AGENT_AUTH="${AUTH_TYPE}"
 export CODER_AGENT_URL="${ACCESS_URL}"
-exec ./$BINARY_NAME agent
+
+output=$(./${BINARY_NAME} --version | head -n1)
```

Comment:
nit, non-blocking: I wonder if someone is going to try running this on a system that doesn't have the `head` or `grep` command? Could we just do a substring match instead?

Reply:
Since we're using `sh` not `bash`, I was having a tough time making this robust. We already depend on `curl` implicitly so it doesn't feel like a big ask, especially since they're included in all *nix OSs by default AFAIK, while `curl` isn't.

I tried https://stackoverflow.com/a/40771648 but that didn't work out IIRC. Open to suggestions here.

Reply:
I agree, it's not a big ask. I'm probably nit-picking at nothing here.

Reply:
Your spidey-senses are on point though; it is a bit smelly.

Reply:
I don't think we need the `head` anyway, `grep` should suffice even on a multi-line output 🤔.

Btw, we could either do this additional check:
```shell
if ! output="$(./bin --version 2>&1)"; then
    error running coder binary... $output
fi
```

Or we could just combine them:
```
if ! ./bin --version 2>&1 | grep -q Coder; then ...; fi
```

Reply:
I used `head -n1` to only select the first line of output to make the diagnostic message more readable (it might be hundreds of lines long otherwise).

Reply:
Ah, I see why you did it then, it's definitely reasonable. It could still obstruct some useful information which is only on the following lines, but it's still better than nothing, so 👍🏻.